policies/branch-policy.md

```diff
@@ -0,0 +1,100 @@
+  usually will) limit the extent of the breakage allowed.
+- Major features are allowed. Examples of a major feature: A completely
+  new implementation of a protocol. New API for pluggable crypto modules.
+- Major refactoring is allowed. Examples of major refactoring: Splitting
+  libcrypto into multiple hierarchically dependent libraries.
+
+## The supported release branches
+
+- Only bug fixes and documentation changes are allowed.
+- The development of the next patch releases of supported minor releases
+  happens there.
+- By exception given by OMC also other types of pull requests can be merged.
+- During the final year of support, we do not commit to anything other than
+  security fixes.
+- Bug fix and documentation pull requests must be always merged to the
+  latest branch (including the master branch) where the bug or documentation
+  deficiency is present. It can be then merged (backported or directly
+  cherry-picked) to all older branches where the deficiency is present.
+
+## A feature branch
+
+- There exists one major feature branch where collective development of
+  a major feature X is happening.
+- Any changes that are needed only for the X feature are merged to this branch.
+- To ensure easy mergeability of the feature branch to master once the
+  development is finished, the branch is periodically (once a month) rebased
+  on top of the master branch. The rebase might happen more often if there is
+  already a larger amount of changes accumulated in the master branch.
+- The review process for the feature branch is the same as the review process
+  for the master branch except for the rebases where only casual review is
+  expected but the CI must pass fully before the force-push of the rebased
+  branch is allowed.
+- Once the development of the feature X is completed the OTC approves merge
+  of the freshly rebased feature branch to the master branch.
+
+## A next minor branch
+
+- The development of the next minor release happens there.
+- No API/ABI breaking changes are allowed.
+- No major features are allowed unless explicitly acked by OMC as targeted for
+  the minor release.
+- No major refactoring is allowed.
+- Any changes (features, bug-fixes, documentation, ...) done on the next-minor
+  branch must be forward-ported or directly cherry-picked to the master branch
+  if applicable.
+- Exceptions are possible for example when we are removing deprecated
+  functionality in the master branch.
+- Backporting of features and refactoring changes from the master branch to the
+  next-minor branch happens on a case-by-case basis decided by OTC. **Can we
+  find criteria for making the process less heavyweight on OTC and more
+  expectable by outside contributors? For example we might say that minor
+  features/refactorings are implicitly allowed but it is up to the PR submitter
+  to say whether he wants the PR to be merged to both master and next-minor
+  or just master.**
```

Comment:
Having the PR submitter indicate their desire seems quite useful.
I am not sure that trying to say "minor refactoring is allowed" will help much, since the boundary between minor/major is subjective.

Reply:
I agree. I do not like these subjective adverbs in policies. And it would be really hard to give some exact limits such as number of changed lines or so because renaming an internal function might give pretty large PR but without any serious destabilization impact.